src/cmake-tools.ts

```diff
@@ -784,6 +783,15 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
       consumer.dispose();
     }
   }
+  /**
+   * Implementation of `cmake.build`
+   */
+  async build(target_?: string): Promise<number> {
+    this.m_promise_build = this.RunBuild(target_);
+    const run_build = this.m_promise_build;
```

Comment:
Do we need these two intermediate variables or can you just return `await run_build` here?

Reply:
It is only possible to have one waiting function on a promise. I had the problem that the build method and the stop build method locked each other. The m_promise_build synchronizes different tasks. The copy of the m_promise ensures that the build function can wait.

Reply:
Promises aren't copied, right?  I was just suggesting trimming some of the code that looked unnecessary.
```ts
  async build(target_?: string): Promise<number> {
    this.m_promise_build = this.RunBuild(target_);
    return await this.m_promise_build;
  }
```
You're saying that that doesn't work?

Reply:
I had some trouble with the `m_promise_build` Promise, but I don´t remember the exact problem. This helper variable does not make sense.